### PR TITLE
✨ 🏷️ 257 - Implement mapper and PHI DAS services for retrieving OhipNumber

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Implements functionality in Data Mapper and PHI DAS for retrieving an OhipNumber by private key, and adds to the `getConsentReleaseData` response for Step 2.

### Data Mapper
- implements `getOhipNumberRecord` function for retrieving ohip number from PHI DAS
- adds helper function in `search` service to retrieve ohip info (private key + number) if `hasOhip` is true

### PHI DAS
- implements `GET /ohip/{ohipPrivateKey}` endpoint
- updates swagger schemas
- adds `getOhipNumberRecord` to search service

### Types
- adds response types for `GetOhipNumberRecord` in dataMapper and phiDas services

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/services/search.ts [hunk 1]
```diff
@@ -279,16 +288,26 @@ export const getConsentReleaseData = async (
...
+		}
 		//** PHI DAS service call for ClinicalProfile */
 
 		// combine all response data and parse
 		const completeResponse = {
 			...piParticipantData.data,
 			...consentQuestionResponses.data,
-			// ...(ohipRecordResponse?.data || {}),
-			// ...updateClinicalProfileResponse.data,
+			// ...clinicalProfile,
```
Comment (anncatton) [inline apps/data-mapper/src/services/search.ts:307]: clinicalProfile data added in next PR https://github.com/OHCRN/platform/pull/599

File: apps/data-mapper/src/services/search.ts [hunk 2]
```diff
@@ -230,6 +233,12 @@ export const getInformedConsentResponse = async (
 	}
 };
 
+type GetConsentReleaseDataFailureStatus =
```
Comment (mistryrn) [inline apps/data-mapper/src/services/search.ts:236]: Nice and clear naming 👍

File: apps/phi-das/src/routers/ohip.ts [hunk 3]
```diff
@@ -80,20 +82,46 @@ router.get('/', async (req, res) => {
...
+ *       404:
+ *         description: NotFoundError - OhipNumber with the specified ohipPrivateKey could not be found.
  *       500:
- *         description: Error retrieving ohip number.
+ *         description: Error retrieving OhipNumber record.
  */
 router.get('/:ohipPrivateKey', async (req, res) => {
-	logger.info('GET /ohip/:ohipPrivateKey');
 	const { ohipPrivateKey } = req.params;
-	// TODO: add validation
```
Comment (mistryrn) [inline apps/phi-das/src/routers/ohip.ts:90]: 🥲 love seeing these TODO's crossed off

File: apps/phi-das/src/services/search.ts [hunk 4]
```diff
@@ -17,16 +17,42 @@
...
  */
 
+import { Result, SystemError, failure, success } from 'types/httpResponses';
+
+import { PrismaClientKnownRequestError } from '../generated/client/runtime/library.js';
 import prisma, { ClinicalProfile, Ohip } from '../prismaClient.js';
+import serviceLogger from '../logger.js';
 
-export const getOhipNumber = async (ohipPrivateKey: string): Promise<Ohip> => {
-	// TODO: add error handling
```
Comment (mistryrn) [inline apps/phi-das/src/services/search.ts:23]: 🎉

**Review by mistryrn — APPROVED** [review]
Good stuff, nice and clean!

**Pull request merged**